### PR TITLE
Recreate slabs after graceful shutdown

### PR DESCRIPTION
**Pull request by michalbiesek**
- USE_PMEM configuration allows to restore hashtable storage

-Continue of PR #3
-Recreate slab_lruq will be done in next PR

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pelikan/4)

### CODE REVIEW

File: src/storage/slab/slab.h [hunk 1]
```diff
@@ -129,7 +129,8 @@ struct slab {
     TAILQ_ENTRY(slab) s_tqe;        /* link in slab lruq */
 
     uint8_t           id;           /* slabclass id */
-    uint32_t          padding:24;   /* unused */
+    uint8_t           initialized;  /* flag is slab initialized*/
+    uint16_t          padding;      /* unused */
```
Comment (pbalcer) [inline src/storage/slab/slab.h]: `uint16_t unused; /* unused, must be 0 */`

And zero this field during initialization.

File: src/storage/slab/slab.c [hunk 2]
```diff
@@ -127,6 +128,91 @@ slab_id(size_t size)
...
+    p = &slabclass[slab->id];
+
+    for (i = 0; i < p->nitem; i++) {
+        it = _slab_to_item(slab, i, p->size);
+        if (it->is_linked) {
+            INCR(slab_metrics, item_curr);
+            INCR(slab_metrics, item_alloc);
+            PERSLAB_INCR(slab->id, item_curr);
+            item_relink(it);
+        }
```
Comment (pbalcer) [inline src/storage/slab/slab.c]: `} else if {`

**Review by pbalcer — APPROVED** [review]

File: src/storage/slab/slab.h (hunk 1 above)
Comment (michalbiesek) [reply src/storage/slab/slab.h]: Done

File: src/storage/slab/slab.c (hunk 2 above)
Comment (michalbiesek) [reply src/storage/slab/slab.c]: Done

**Review by jschmieg — APPROVED** [review]

**Pull request merged**